harvester_e2e_tests/integrations/test_upgrade.py

```diff
@@ -872,59 +895,199 @@ def test_degraded_volume(self, cluster_prereq, request, api_client, cluster_stat
+        api_client.upgrades.delete(data['metadata']['name'])
+        api_client.versions.delete(version)
+
+    @pytest.mark.skip("https://github.com/harvester/harvester/issues/5494")
+    def test_version_compatibility(
+        self, api_client, unique_name, upgrade_target, upgrade_timeout
+    ):
+        version, url, checksum = upgrade_target
+        version = f"{version}-{unique_name}"
+
+        code, data = api_client.versions.create(version, url, checksum)
+        assert 201 == code, f"Failed to create upgrade for {version}"
+        code, data = api_client.upgrades.create(version)
+        assert 201 == code, f"Failed to start upgrade for {version}"
+
+        endtime = datetime.now() + timedelta(seconds=upgrade_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.upgrades.get(data['metadata']['name'])
+            conds = dict((c['type'], c) for c in data.get('status', {}).get('conditions', []))
+            verified = []  # TODO
+            if all(verified):
+                break
+        else:
+            raise AssertionError(f"Upgrade NOT failed in expected conditions: {conds}")
+
+        # teardown
+        api_client.upgrades.delete(data['metadata']['name'])
+        api_client.versions.delete(version)
 
 
 @pytest.mark.upgrade
 @pytest.mark.any_nodes
 class TestAnyNodesUpgrade:
 
     @pytest.mark.dependency(name="any_nodes_upgrade")
-    def test_perform_upgrade(self, cluster_prereq, vm_prereq, request, api_client, cluster_state,
-                             wait_timeout):
+    def test_perform_upgrade(
+        self, cluster_prereq, vm_prereq, api_client, cluster_state, wait_timeout,
+        unique_name, upgrade_target, upgrade_timeout, interceptor
+    ):
         """
         - perform upgrade
         - check all nodes upgraded
         """
-        if cluster_state.version_verify:
-            assert not _is_installed_version(api_client, cluster_state.version), (
-                f"The current version is already {cluster_state.version}")
+        # Check nodes counts
+        code, data = api_client.hosts.get()
+        assert code == 200, (code, data)
+        nodes = len(data['data'])
+
+        # create Upgrade version and start
+        version, url, checksum = upgrade_target
+        version = f"{version}-{unique_name}"
+        code, data = api_client.versions.create(version, url, checksum)
+        assert 201 == code, f"Failed to create upgrade for {version}"
+        code, data = api_client.upgrades.create(version)
+        assert 201 == code, f"Failed to start upgrade for {version}"
+        upgrade_name = data['metadata']['name']
 
-        self._perform_upgrade(request, api_client, cluster_state)
+        # Check upgrade status
+        # TODO: check every upgrade stages
+        endtime = datetime.now() + timedelta(seconds=upgrade_timeout * nodes)
+        while endtime > datetime.now():
+            code, data = api_client.upgrades.get(upgrade_name)
```

Comment:
A question.
Since nodes will experiencing reboot during upgrade.
Do we ever hit `api_client` connection broken during cluster upgrade?

Reply:
I didn't encounter the case, I feel that the server will return `500` when VIP is not working.